skfuzzy/control/fuzzyvariable.py

```diff
@@ -44,12 +82,30 @@ def __init__(self, universe, label):
             Unique name of the universe variable, e.g., 'food' or 'velocity'.
         """
         self.universe = np.asarray(universe)
-        self.active = None
-        self.mf = OrderedDict()
         self.label = label
-        self.connections = OrderedDict()
+        self.defuzzy_method = defuzzy_method
+        self.adjectives = OrderedDict()
+
         self._id = id(self)
-        self.output = None
+        self._crisp_value_accessed = False
+
+        class _NotGenerator(object):
```

Comment:
users should not be using variable['NOT-adjective'], they should use variable.not_['adjective'].